server/src/main/java/org/opensearch/cluster/routing/allocation/ExistingShardsAllocator.java

```diff
@@ -60,6 +60,13 @@ public interface ExistingShardsAllocator {
         Setting.Property.PrivateIndex
     );
 
+    public static final Setting<Boolean> EXISTING_SHARDS_ALLOCATOR_BATCH_MODE_ENABLED = Setting.boolSetting(
+        "cluster.allocator.existing_shards_allocator.batch_enable",
+        true,
+        Setting.Property.NodeScope
```

Comment:
Add `Property.Dynamic` here ?

Reply:
Got it. Any reason the default value is `false` here ?

Reply:
Other PRs in the issue highlight in the description are dependent on this setting. So we can merge this PR and later other PRs can use this setting easily.

We will have a final PR once all PRs are merge to enable it.

Reply:
"batch_enabled"